ktor-server/jvm/src/io/ktor/server/application/plugins/api/PluginExceptions.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.server.application.plugins.api
+
+/**
+ * Reports that [pluginName] is not installed into Ktor server but is required by some other plugin.
+ */
+public class PluginNotInstalledException(private val pluginName: String) : Exception() {
+    override val message: String
+        get() = "Plugin $pluginName is not installed but required"
+}
+
+/**
+ * Reports that current [subject] in HTTP pipeline is not of an expected binary data type.
+ */
+public open class NoBinaryDataException(private val expectedTypeName: String, private val subject: Any?) : Exception() {
```

Comment:
Let's have a single exception: `InvalidBodyException` with clarification in the message. It's hard to catch all of them separately and there is no use case for that